elastic/kibana
# [Connection Details] Check if user has permissions to manage own API keys

## Summary

Closes https://github.com/elastic/kibana/issues/181288

- Exposes from `security` plugin ability to check if use can manage own API keys, through `security.authz.getCurrentUserApiKeyPrivileges()`.
- Checks if user has permissions, and displays "No permissions" screen, if user does not have permissions in the Connection Details flyout.

## How to test

Add these Cloud settings to your `config/kibana.dev.yml` as a mock data:

```
elasticsearch.hosts: ["http://localhost:9200"]
xpack.cloud.id: my-cluster-id:dXMtZWFzdC0xLmF3cy5zdGFnaW5nLmZvdW5kaXQubm8kZjY3ZDZiZjFhM2NmNDA4ODhlODg2M2Y2Y2IyY2RjNGMkOWViYzEzYjRkOTU0NDI2NDljMzcwZTNlZjMyZWYzOGI=
```

This will expose the "Setup guides" button in the top nav and the "Connect to the Elasticsearch API" card:

![image](https://github.com/elastic/kibana/assets/82822460/c1d5a657-183f-4af6-90c3-c1d393379785)

Create a new user. The lowest level of API key permissions is `manage_own_api_key`. Toggle that permission on user settings to see the changes in the Connection Details flyout.

<img width="1232" alt="Screenshot 2024-05-13 at 15 27 41" src="https://github.com/elastic/kibana/assets/82822460/c3b65a25-4b7e-4215-909b-4ad804959a2b">


When the user does not have a permission to edit own API keys, the connection details flyout should show this message:

<img width="464" alt="image" src="https://github.com/elastic/kibana/assets/82822460/4916bbbb-ded9-4c6c-9c49-ca5ce56e320c">


When user has permissions, it should allow to create a new API key:

<img width="460" alt="image" src="https://github.com/elastic/kibana/assets/82822460/2839908c-4fb6-4bcc-aeb4-bea1205526df">




### Checklist

Delete any items that are not applicable to this PR.



### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers
